File: Leetcode/easy/9.cpp
#include <vector>

class Solution {
public:
    bool isPalindrome(int x) {
        vector<int> arr;
        
        if (x < 0) return false;
        
        while (x) {
            int value = x % 10;
            arr.push_back(value);
            x = x / 10;
        }
        
        for (int i = 0; i < arr.size(); ++i) {
            if (arr[i] != arr[arr.size() - i - 1]) {
                return false;
            }
        }
        
        return true;
    }
};

File: Leetcode/easy/9.swift
class Solution {
    func isPalindrome(_ x: Int) -> Bool {
        var str: String = String(x)
        
        if x < 0 {
            return false
        }

        for i in 0..<str.length {
            if str[str.index(str.startIndex, offsetBy: i)] != str[str.index(str.endIndex, offsetBy: -i-1)] {
                return false
            }
        }
        
        return true
    }
}